packages/button/components/ButtonBase.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react";
+import { cx } from "emotion";
+import {
+  buttonReset,
+  textWeight,
+  flex,
+  flexItem,
+  display,
+  padding
+} from "../../shared/styles/styleUtils";
+import {
+  button,
+  buttonBase,
+  buttonContent,
+  buttonInverse,
+  fullWidthButton,
+  getInverseMutedButtonStyles,
+  getMutedButtonStyles,
+  processingText
+} from "../style";
+
+export enum ButtonAppearances {
+  Primary = "primary",
+  Secondary = "secondary",
+  Standard = "standard",
+  Danger = "danger",
+  Success = "success"
+}
+
+export interface ButtonProps {
+  /**
+   * if the button triggers new content to appear (e.g.: modals and dropdowns)
+   */
+  ariaHaspopup?: boolean;
+  children?: React.ReactNode | string;
+  /**
+   * whether or not the button is enabled
+   */
+  disabled?: boolean;
+  /**
+   * the icon that appears before the button text
+   */
+  iconStart?: React.ReactElement<HTMLElement>;
+  /**
+   * the icon that appears after the button text
+   */
+  iconEnd?: React.ReactElement<HTMLElement>;
+  /**
+   * whether the action the button was meant to do has completed
+   */
+  isProcessing?: boolean;
```

Comment:
Isn't `isActive` a better name here?

Reply:
I prefer this-- active could be confused for what happens while the mouse is down

Reply:
Agreed with Brandon, and "processing" is how the design team refers to that state, and it would be nice if we could use shared language

Reply:
Okay sounds good to me.